niceman/retrace/packagemanagers.py

```diff
@@ -273,29 +280,24 @@ def _create_package(self, pkgname):
         lgr.debug("Found package %s", pkg)
         return pkg
 
-    def _find_release_date(self, site, archive):
-        if not site:
-            return None
-        # First find the release file
-        # This uses logic similar to apt.utils.get_release_filename_for_pkg
-        # but restricts matches to the source site
+    def _find_release_file(self, package_filename):
+        # The release filename is a substring of the package
+        # filename (excluding the ending "Release" or "InRelease"
+        # The split between the release filename and the package filename
+        # is at an underscore, so split the package filename
+        # at underscores and test for the release file:
+        rfprefix = package_filename
         rfile = None
-        dirname = apt_pkg.config.find_dir("Dir::State::lists")
-        # If we want to avoid using cache._list, we can call
-        # apt_pkg.SourceList() directly
-        for uri in set([metaindex.uri for metaindex in cache._list.list]):
-            if site == urlparse(uri).netloc:
-                for relfile in ['InRelease', 'Release']:
-                    name = ((apt_pkg.uri_to_filename(uri)) +
-                            "dists_%s_%s" % (archive, relfile))
-                    if os.path.exists(opj(dirname, name)):
-                        if rfile:
-                            raise MultipleReleaseFileMatch(
-                                "More than one release file found for %s %s" %
-                                (site, archive))
-                        rfile = opj(dirname, name)
-
-        # Now extract and format the date from the release file
+        while "_" in rfprefix:
+            rfprefix = rfprefix.rsplit("_", 1)[0]
+            for ending in ['_InRelease', '_Release']:
+                if os.path.exists(rfprefix + ending):
+                    rfile = rfprefix + ending
+                    break
```

Comment:
I think here a bit of an odd and rarely used python feature could help to avoid duplication

```python
for ending in ['_InRelease', '_Release']:
    rfile = rfprefix + ending
    if os.path.exists(rfile):
        break
else:
     rfile = None
```

or if you want a really Pythonic one (never used such but stackoverflow came to help):
```python
return next((p for p in [rprefix+e for e in ['_InRelease', '_Release']] 
                    if os.path.exists(p)),
                   None)
```
which with a little bit of help becomes actually easy to read one liner (ignore import, needs to be adjusted to use six):
```python
from itertools import ifilter
return next(ifilter(os.path.exists, map(rprefix.__add__, ['_InRelease', '_Release'])), None)
```
;-)